### PR TITLE
CDK Go Sample - ECS Cluster With ALB

### LINKED ISSUE #811: Expand CDK Go Samples
### Describe the feature

When reviewing the AWS CDK Go samples, I noticed there weren't many present. Hopefully this issue will help drive adoption for using Go with CDK through clear samples. 

### Use Case

The goal of this issue if to increase the number of samples for Go to assist those who may be new to either Go or CDK.

### Proposed Solution

I can assist in creating new CDK Go samples as I have experience with both.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Language

Go

### PR DESCRIPTION
**Pull request by yaseenyounus**
# Overview
* Adds a new CDK Go sample that demonstrates how to configure and deploy an ECS Cluster with an Application Load Balancer (ALB). 
* Provides a README with deployment and testing guidance.
* Contains custom Go tests for the infrastructure / stack being deployed.

---

Fixes #811 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: go/ecs/cluster-alb/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- `cdk diff` to compare local changes with what is currently deployed.
+- `go test` to run the tests we specify in `cluster-alb_test.go`.
+- `cdk deploy` to deploy the stack to the AWS account you're authenticated to.
+
+## Testing
+
+- CDK will output the ALB endpoint. Copy/paste it into the browser to check that the application is running.
+  - e.g. `http://EcsClusterALB-xxxxxxxxx.us-east-1.elb.amazonaws.com`
+  - _Note: Be sure to prefix the ALB endpoint with `http://` as your browser may initially force `https`, which will not work as we are not issuing or installing an SSL certificate in this sample._
+- Navigate to the AWS console and view the services that were deployed in the VPC (Subnets, Route Tables, etc), ECS (Cluster, Service, Task Definition, etc) and EC2 (Load Balancer, Target Groups, etc) consoles.
```
Comment (indrora) [inline go/ecs/cluster-alb/README.md:35]: * How can the customer see that the ALB is being used? 
* How can the customer enable TLS?
* What actions can the customer take to extend this?

File: go/ecs/cluster-alb/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+---
+
+<!--END STABILITY BANNER-->
+
+## Overview
+
+This AWS CDK Go sample demonstrates how to configure and deploy an Elastic Container Service (ECS) Cluster behind an Application Load Balancer (ALB).
+
+This is useful because traffic will not directly hit the ECS cluster; rather, it will be routed through the ALB to access the application.
```
Comment (indrora) [inline go/ecs/cluster-alb/README.md:19]: Give a real-world example of where this would make sense to use.

**Review by indrora — CHANGES_REQUESTED** [review]
Improve the readme and it looks good to me.

**Comment by yaseenyounus** [comment]
> Improve the readme and it looks good to me.

Thanks for the suggestions @indrora. Please let me know your thoughts on the README updates. Thanks!

File: go/ecs/cluster-alb/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+<!--END STABILITY BANNER-->
+
+## Overview
+
+This AWS CDK Go sample demonstrates how to configure and deploy an Elastic Container Service (ECS) Cluster behind an Application Load Balancer (ALB).
+
+## Real-world Example
+
+Imagine we are running a containerized application with multiple tasks within an ECS Cluster service. We wouldn't want to expose the service directly to the internet as part of security best practices. Therefore, we deploy an Application Load Balancer (ALB) in front of the service to both balance traffic, and act as the single point of entry to the application.
```
Comment (indrora) [inline go/ecs/cluster-alb/README.md:21]:
```suggestion
When running a containerized application with multiple tasks in an ECS Cluster service, not exposing the service directly to the internet as part of security best practices. In this example, we deploy an Application Load Balancer (ALB) in front of the service to both balance traffic, and act as the single point of entry to the application.
```

File: go/ecs/cluster-alb/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- Navigate to the AWS console to view the services that were deployed:
+  - VPC (Subnets, Route Tables, etc.)
+  - ECS (Cluster, Service, Task Definition, etc.)
+  - EC2 (Load Balancer, Target Groups, etc.)
+
+## Further Improvements
+
+- Enable TLS at the ALB by [creating an HTTPS listener](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#add-https-listener).
+- Route traffic to the ALB with a [custom Route 53 domain](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-to-elb-load-balancer.html).
+- Protect your ALB from attacks and malicious traffic using [Web Application Firewall (WAF)](https://docs.aws.amazon.com/waf/latest/developerguide/waf-chapter.html).
```
Comment (indrora) [inline go/ecs/cluster-alb/README.md:61]:
```suggestion
## Further Improvements

This example does not use an HTTPS listener. For improved security, HTTPS (TLS) is encouraged. Some further improvements that can be made are:

- Enable TLS at the ALB by [creating an HTTPS listener](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#add-https-listener).
- Use a [CloudFront distribution to cache results returned from the application](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/DownloadDistS3AndCustomOrigins.html#concept_elb_origin), reducing load on the containerized application overall.
- Route traffic to the ALB with a [custom Route 53 domain](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-to-elb-load-balancer.html).
- Protect your ALB from attacks and malicious traffic using [Web Application Firewall (WAF)](https://docs.aws.amazon.com/waf/latest/developerguide/waf-chapter.html).
```

**Review by indrora — DISMISSED** [review]
Some suggestions to clear up the language. 

I'll rubber stamp these minor edits.

**Review by indrora — APPROVED** [review]

**Review by indrora — APPROVED** [review]

**Pull request merged**